scripts/nfs-export-updater

```diff
@@ -25,21 +27,83 @@ from nfs_export_updater.server import NFSRootUpdateServer
 logger = logging.getLogger('nfs-export-updater')
 
 
+def create_nfs_exportdir(rootfs, extract_dir):
+    unpack_archive = None
+
+    import bb.tinfoil
+    with bb.tinfoil.Tinfoil() as tinfoil:
+        tinfoil.prepare(quiet=1)
+
+        rd = tinfoil.parse_recipe(rootfs)
+        src_prefix = '%s/%s' % (rd.getVar('DEPLOY_DIR_IMAGE'), rd.getVar('IMAGE_LINK_NAME'))
+        for suffix in ['.tar.bz2', '.tar.gz']:
+            check_path  = f"{src_prefix}{suffix}"
+            logging.debug(f"checking archive path {check_path}")
+            if (os.path.exists(check_path)):
+                unpack_archive = check_path
+                logging.debug(f"Found existing archive at {check_path}")
+                break
+
+    if not unpack_archive:
+        logger.error(f"Unable to unpack: Could not find rootfs archive. Did you run 'bitbake {rootfs}'?")
+        sys.exit(1)
+
+    cmd = ('runqemu-extract-sdk', unpack_archive, extract_dir)
+    logger.info('Running %s...' % str(cmd))
+    if subprocess.call(cmd) != 0:
+        raise Exception('Failed to run %s' % str(cmd))
+
+
+def setup_nfs_exporter(rootfs):
+    nfs_instance = 0
+
+    # Figure out a new nfs_instance to allow multiple qemus running.
+    ps = subprocess.check_output(("ps", "auxww")).decode('utf-8')
+    pattern = '/bin/unfsd .* -i .*\.pid -e .*/exports([0-9]+) '
+    all_instances = re.findall(pattern, ps, re.M)
+    if all_instances:
+        all_instances.sort(key=int)
+        nfs_instance = int(all_instances.pop()) + 1
+
+    nfsd_port = 3049 + 2 * nfs_instance
+    mountd_port = 3048 + 2 * nfs_instance
```

Comment:
This was copied from an old version of `scripts/runqemu`. This should be adapted to https://github.com/yoctoproject/poky/commit/011e2bb251.

Reply:
I have added the new upstream version but omitted the locking for now since it would involve copying more methods.